# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1303842448 25200
#      Tue Apr 26 11:27:28 2011 -0700
# Branch stable
# Node ID d02798e9eeaec12825fd5b386756ea9195d69102
# Parent  accd03f6698256dd21d0d374dcd6097ee32ac7a6
filter.cc: Stop test errors in residue.m (bug #33164)

* filter.cc: Reshape b,a inputs to column vectors

diff --git a/src/DLD-FUNCTIONS/filter.cc b/src/DLD-FUNCTIONS/filter.cc
--- a/src/DLD-FUNCTIONS/filter.cc
+++ b/src/DLD-FUNCTIONS/filter.cc
@@ -59,16 +59,24 @@ filter (MArray<T>& b, MArray<T>& a, MArr
 {
   MArray<T> y;
 
   octave_idx_type a_len  = a.length ();
   octave_idx_type b_len  = b.length ();
 
   octave_idx_type ab_len = a_len > b_len ? a_len : b_len;
 
+  // FIXME: The two lines below should be unecessary because
+  //        this template is called with a and b as column vectors
+  //        already.  However the a.resize line is currently (2011/04/26)
+  //        necessary to stop bug #33164.
+  b.resize (dim_vector (ab_len, 1), 0.0);
+  if (a_len > 1)
+    a.resize (dim_vector (ab_len, 1), 0.0);
+
   T norm = a (0);
 
   if (norm == static_cast<T>(0.0))
     {
       error ("filter: the first element of A must be non-zero");
       return y;
     }
 
